maven-archetype/src/main/resources/META-INF/archetype-post-generate.groovy

```diff
@@ -0,0 +1,22 @@
+import java.nio.file.Files
+import java.nio.file.Paths
+
+projectDirectory = Paths.get(request.outputDirectory, request.artifactId);
+
+def run(String command) {
+    def process = command.execute(null, projectDirectory.toFile());
+    process.consumeProcessOutput(System.out, System.err)
+    process.waitFor()
+    if (process.exitValue() != 0) {
+        throw new RuntimeException("'$command' exited with code ${process.exitValue()}")
+    }
+}
+
+// add the Maven wrapper
+run("mvn wrapper:wrapper")
```

Comment:
What version of Maven is running here? Does this assume Maven to be installed on the machine?

The fact that the project build log looses the coloring when running this command made me wonder...
<img width="1270" alt="Screenshot 2024-02-12 at 07 59 43" src="https://github.com/ozone-his/ozone/assets/10692811/ac5fccfa-33e0-48fd-971a-6ba4f6b97ec3">

Would it be possible to use the Wrapper provided by the project to run the commands instead?

Reply:
So, it turns out to be pretty hard to do this. If I _just_ do this with the wrapper from this repo, then the script won't work when used outside of this repo (it'd also get pretty gnarly to make integration test pass). I was hoping to be able to get access to the Maven session, so I could just execute things in the current Maven context, but that isn't available inside the script.

I've resorted to the work-around where I've basically re-implemented the `mvn wrapper:wrapper` part in Groovy, using Groovy built-ins. Then, once the wrapper is installed, we use that for any additional Maven commands. The script is a little verbose, but it should basically work.